packages/@aws-cdk/service-spec-importers/src/db-builder.ts

```diff
@@ -0,0 +1,188 @@
+import { emptyDatabase, SpecDatabase } from '@aws-cdk/service-spec-types';
+import { assertSuccess, Result } from '@cdklabs/tskb';
+import { Augmentations } from './importers/import-augmentations';
+import { importCannedMetrics } from './importers/import-canned-metrics';
+import { importCloudFormationDocumentation } from './importers/import-cloudformation-docs';
+import { importCloudFormationRegistryResource } from './importers/import-cloudformation-registry';
+import { ResourceSpecImporter, SAMSpecImporter } from './importers/import-resource-spec';
+import { SamResources } from './importers/import-sam';
+import { Scrutinies } from './importers/import-scrutinies';
+import { importStatefulResources } from './importers/import-stateful-resources';
+import {
+  loadDefaultCloudFormationDocs,
+  loadDefaultCloudFormationRegistryResources,
+  loadDefaultCloudWatchConsoleServiceDirectory,
+  loadDefaultResourceSpecification,
+  loadDefaultStatefulResources,
+  LoadResult,
+  loadSamSchema,
+  loadSamSpec,
+} from './loaders';
+import { ProblemReport, ReportAudience } from './report';
+
+export interface DatabaseBuilderOptions {
+  /**
+   * Fail if we detect schema validations with the data source
+   * @default true
+   */
+  readonly validate?: boolean;
+
+  /**
+   * Print additional debug information
+   * @default false
+   */
+  readonly debug?: boolean;
+}
+
+export type SourceImporter = (db: SpecDatabase, report: ProblemReport) => Promise<void>;
+
+export class DatabaseBuilder {
+  private readonly sourceImporters = new Array<SourceImporter>();
```

Comment:
Why is this lazy?

Reply:
Mostly to improve UX, this enables a fluent interface. Otherwise it's a bunch of statements that need to be awaited.
I don't have strong feelings about this. It made sense to me to expose the importers as methods on a class.